src/core/component-base.ts

```diff
@@ -172,7 +169,7 @@ abstract class ComponentBase<P extends IHtmlOptions> extends React.PureComponent
   private _prepareProps(rawProps: Record<string, any>): IWidgetConfig {
     const separatedProps = separateProps(rawProps, this._defaults, this._templateProps);
 
-    this._templatesManager.add({
+    const templatesOptions = this._templatesManager.add({
```

Comment:
It's not clear what does the `add` method return. I mean a method `called` is `void` mostly, can this one be split into a command and a query?

Reply:
Looks like it requires complex changes and I would suggest to leave this for further improvements.

Reply:
Ok